backend/mora/common.py

```diff
@@ -50,6 +51,77 @@ def get(self, objid):
         pass
 
 
+class RequestType(enum.Enum):
+    CREATE = 0
+    EDIT = 1
+
+
+class Request(abc.ABC):
+
+    __slots__ = 'request', 'request_type', 'payload', 'uuid'
+
+    def __init__(self, request, request_type: RequestType):
+        """
+        Initialize a request, and perform all required validation.
+        :param request: A dict containing a request
+        :param request_type: A RequestType, either CREATE or EDIT
+        """
+        self.request_type = request_type
+        self.request = request
+        self.payload = None
+        self.uuid = None
+
+        if request_type == RequestType.CREATE:
+            self.create(request)
+        if request_type == RequestType.EDIT:
+            self.edit(request)
+
+    @abc.abstractmethod
+    def create(self, request: dict):
+        """
+        Initialize a 'create' request. Performs validation and all
+        necessary processing
+        :param request: A dict containing a request
+        """
+        pass
+
+    @abc.abstractmethod
+    def edit(self, request: dict):
+        """
+        Initialize an 'edit' request. Performs validation and all
+        necessary processing
+        :param request: A dict containing a request
+        """
+        pass
+
+    @abc.abstractmethod
+    def submit_request(self) -> str:
+        """
+        Submit the request to LoRa.
+        :return: A string containing the result from submitting the request
+        to LoRa, typically a UUID.
+        """
+        pass
+
+
+class OrgFunkRequest(Request):
+    @abc.abstractmethod
+    def create(self, request: dict):
```

Comment:
The naming is a bit off here; `create` implies that the method creates something when in fact it does not. `prepare_creation`?